Remove warnings in samples which appear in the Kokoro logs

```
__________________________ test_storage_compose_file ___________________________

test_bucket = 

    def test_storage_compose_file(test_bucket):
        source_files = ["test_upload_blob_1", "test_upload_blob_2"]
        for source in source_files:
            blob = test_bucket.blob(source)
            blob.upload_from_string(source)
    
        with tempfile.NamedTemporaryFile() as dest_file:
            destination = storage_compose_file.compose_file(
                test_bucket.name,
                source_files[0],
                source_files[1],
                dest_file.name,
            )
>           composed = destination.download_as_string()
....
E       PendingDeprecationWarning: Blob.download_as_string() is deprecated and will be removed in future. Use Blob.download_as_bytes() instead.
```


```
________________________ ERROR at setup of test_get_rpo ________________________

    @pytest.fixture
    def dual_region_bucket():
        """Yields a dual region bucket that is deleted after the test completes."""
        bucket = None
        while bucket is None or bucket.exists():
            bucket_name = f"bucket-lock-{uuid.uuid4()}"
            bucket = storage.Client().bucket(bucket_name)
>           bucket.location = "NAM4"

rpo_test.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = 
value = 'NAM4'

    @location.setter
    def location(self, value):
        """(Deprecated) Set `Bucket.location`
    
        This can only be set at bucket **creation** time.
    
        See https://cloud.google.com/storage/docs/json_api/v1/buckets and
        https://cloud.google.com/storage/docs/bucket-locations
    
        .. warning::
    
            Assignment to 'Bucket.location' is deprecated, as it is only
            valid before the bucket is created. Instead, pass the location
            to `Bucket.create`.
        """
>       warnings.warn(_LOCATION_SETTER_MESSAGE, DeprecationWarning, stacklevel=2)
E       DeprecationWarning: Assignment to 'Bucket.location' is deprecated, as it is only valid before the bucket is created. Instead, pass the location to `Bucket.create`.
```

--- a/docs/snippets.py
+++ b/docs/snippets.py
@@ -39,7 +39,7 @@ def storage_get_started(to_delete):
     bucket = client.get_bucket("bucket-id-here")
     # Then do other things...
     blob = bucket.get_blob("/remote/path/to/file.txt")
-    assert blob.download_as_string() == b"My old contents!"
+    assert blob.download_as_bytes() == b"My old contents!"
     blob.upload_from_string("New contents!")
     blob2 = bucket.blob("/remote/path/storage.txt")
     blob2.upload_from_filename(filename="/local/path.txt")

--- a/samples/snippets/encryption_test.py
+++ b/samples/snippets/encryption_test.py
@@ -125,4 +125,4 @@ def test_object_csek_to_cmek(test_blob):
         BUCKET, test_blob_name, TEST_ENCRYPTION_KEY_2, KMS_KEY
     )
 
-    assert cmek_blob.download_as_string(), test_blob_content
+    assert cmek_blob.download_as_bytes(), test_blob_content

--- a/samples/snippets/rpo_test.py
+++ b/samples/snippets/rpo_test.py
@@ -26,12 +26,13 @@ import storage_set_rpo_default
 @pytest.fixture
 def dual_region_bucket():
     """Yields a dual region bucket that is deleted after the test completes."""
+    storage_client = storage.Client()
     bucket = None
+    location = "NAM4"
     while bucket is None or bucket.exists():
         bucket_name = f"bucket-lock-{uuid.uuid4()}"
-        bucket = storage.Client().bucket(bucket_name)
-        bucket.location = "NAM4"
-    bucket.create()
+        bucket = storage_client.bucket(bucket_name)
+    storage_client.create_bucket(bucket, location=location)
     yield bucket
     bucket.delete(force=True)
 

--- a/samples/snippets/rpo_test.py
+++ b/samples/snippets/rpo_test.py
@@ -26,13 +26,12 @@ import storage_set_rpo_default
 @pytest.fixture
 def dual_region_bucket():
     """Yields a dual region bucket that is deleted after the test completes."""
-    storage_client = storage.Client()
     bucket = None
     location = "NAM4"
     while bucket is None or bucket.exists():
         bucket_name = f"bucket-lock-{uuid.uuid4()}"
-        bucket = storage_client.bucket(bucket_name)
-    storage_client.create_bucket(bucket, location=location)
+        bucket = storage.Client().bucket(bucket_name)
+    bucket.create(location=location)
     yield bucket
     bucket.delete(force=True)
 

--- a/samples/snippets/storage_create_bucket_turbo_replication.py
+++ b/samples/snippets/storage_create_bucket_turbo_replication.py
@@ -37,7 +37,7 @@ def create_bucket_turbo_replication(bucket_name):
     bucket = storage_client.bucket(bucket_name)
     bucket_location = "NAM4"
     bucket.rpo = RPO_ASYNC_TURBO
-    storage_client.create_bucket(bucket, location=bucket_location)
+    bucket.create(location=bucket_location)
 
     print(f"{bucket.name} created with the recovery point objective (RPO) set to {bucket.rpo} in {bucket.location}.")
 

--- a/samples/snippets/storage_create_bucket_turbo_replication.py
+++ b/samples/snippets/storage_create_bucket_turbo_replication.py
@@ -35,9 +35,9 @@ def create_bucket_turbo_replication(bucket_name):
 
     storage_client = storage.Client()
     bucket = storage_client.bucket(bucket_name)
-    bucket.location = "NAM4"
+    bucket_location = "NAM4"
     bucket.rpo = RPO_ASYNC_TURBO
-    bucket.create()
+    storage_client.create_bucket(bucket, location=bucket_location)
 
     print(f"{bucket.name} created with the recovery point objective (RPO) set to {bucket.rpo} in {bucket.location}.")
 

--- a/tests/system/test_blob.py
+++ b/tests/system/test_blob.py
@@ -761,7 +761,7 @@ def test_blob_upload_download_crc32_md5_hash(
 
     download_blob = shared_bucket.blob("MyBuffer")
 
-    assert download_blob.download_as_string() == payload
+    assert download_blob.download_as_bytes() == payload
     assert download_blob.crc32c == blob.crc32c
     assert download_blob.md5_hash == blob.md5_hash
 

--- a/tests/system/test_bucket.py
+++ b/tests/system/test_bucket.py
@@ -410,9 +410,9 @@ def test_bucket_copy_blob_w_metageneration_match(
 ):
     payload = b"DEADBEEF"
     bucket_name = _helpers.unique_name("generation-match")
-    created = _helpers.retry_429_503(storage_client.create_bucket)(
-        bucket_name, requester_pays=True
-    )
+    bucket = storage_client.bucket(bucket_name)
+    bucket.requester_pays = True
+    created = _helpers.retry_429_503(storage_client.create_bucket)(bucket)
     buckets_to_delete.append(created)
     assert created.name == bucket_name
 

